xunit.runner.visualstudio.desktop/VsTestRunner.cs

```diff
@@ -646,20 +732,26 @@ class DiscoveredTestCase
 
             public TestCase VSTestCase { get; }
 
-            string uniqueID;
+            public ITestCase TestCase { get; }
+
+            public string UniqueId { get; }
 
-            public DiscoveredTestCase(string source, ITestFrameworkDiscoverer discoverer, ITestCase testCase, LoggerHelper logger, bool designMode)
+            public DiscoveredTestCase(string source, ITestFrameworkDiscoverer discoverer, ITestCase testCase, LoggerHelper logger, TestPlatformContext testPlatformContext)
             {
                 Name = $"{testCase.TestMethod.TestClass.Class.Name}.{testCase.TestMethod.Method.Name}";
-                TraitNames = testCase.Traits.Keys;
-                VSTestCase = VsDiscoverySink.CreateVsTestCase(source, discoverer, testCase, forceUniqueName: false, logger: logger, designMode: designMode);
-                uniqueID = testCase.UniqueID;
+                TraitNames = testCase.Traits?.Keys;
+                VSTestCase = VsDiscoverySink.CreateVsTestCase(source, discoverer, testCase, forceUniqueName: false, logger: logger, testPlatformContext: testPlatformContext);
+
+                // PERF: note that TestCase is a proxy to actual object in the discovery appdomain. UniqueId is used multiple times,
+                // hence cached to reduce cross-appdomain access cost.
+                TestCase = testCase;
+                UniqueId = testCase.UniqueID;
```

Comment:
We use UniqueID several times later, create a copy once since it is a cross domain access.